### PR TITLE
Add latency controls to new track defaults

### PR DESCRIPTION
**Pull request by SanderVocke**
## Summary
- add recording-alignment and processor-latency controls to the Add Track dialog and apply them during track creation
- register matching persistent controls under Track defaults
- add a `make default` checkbox that saves the Add Track channel, MIDI, and latency values for future drafts
- document the new settings keys and cover UI, persistence, and backend creation behavior

## Tests
- `SHOOP_ALLOW_MISSING_BACKENDS=1 cargo nextest run --workspace --features shoop_engine/app_backend --profile ci` (1616 passed, 4 skipped)
- `RUSTFLAGS="-D warnings" cargo build --workspace`
- `cargo fmt --all -- --check`
- `python3 scripts/check_shoop_test_usage.py`
- `python3 scripts/check_tracing_coverage.py --require-closed`
- `cargo check -p shoopdaloop --no-default-features --target wasm32-unknown-unknown`
- `cargo build -p shoop_audio_worklet --target wasm32-unknown-unknown --release`

The optional Node Wasm package run could not start because the development shell provides Node 22.23.2 while the test runner pins 22.22.2.

### CODE REVIEW

File: src/rust/shoop_app_api/src/lib.rs [hunk 1]
```diff
@@ -1337,10 +1337,19 @@ pub struct AppSnapshot {
...
+    pub manual_frames: i32,
+    pub processor_adjustment: ProcessorLatencyAdjustmentState,
+    pub processor_manual_frames: i32,
+}
+
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct TrackSpec {
     pub name: String,
     pub topology: TrackSpecTopology,
+    pub latency: TrackLatencySpec,
```
[the remark this answers is not in the document]
Comment (SanderVocke) [reply src/rust/shoop_app_api/src/lib.rs:1361]: Addressed in 088b4869. `TrackSpec::validate` now validates active recording and processor latency values against the shared latency domain before the dialog can close. The persisted frame settings are also bounded to ±768,000, and tests cover out-of-range recording values, negative manual processor latency, trim bounds, and preserving an invalid dialog draft.

File: src/rust/shoop_egui/src/app_widget.rs [hunk 2]
```diff
@@ -1719,16 +1989,65 @@ impl AppWidget {
...
 
+    fn accept_add_track_with_defaults(
+        &mut self,
+        processors: &[TrackProcessorDescriptor],
+        settings_state: &SettingsViewState,
+    ) -> Option<(AppAction, Option<SettingsAction>)> {
+        let action = self.accept_add_track(processors)?;
+        let defaults = self
+            .add_track_make_default
+            .then(|| SettingsAction::Save(self.add_track_defaults_draft(settings_state)));
```
[the remark this answers is not in the document]
Comment (SanderVocke) [reply src/rust/shoop_egui/src/app_widget.rs]: Addressed in 088b4869. A make-default creation now carries a correlation ID, and the application publishes a bounded success/failure result only after backend creation is confirmed or rejected. The widget keeps the defaults draft pending, saves it only on success, and discards it without saving on failure. Tests cover both confirmation and rejection paths.

File: src/rust/shoop_egui/src/app_widget.rs [hunk 3]
```diff
@@ -95,6 +110,68 @@ pub fn register_settings_with_appearance_defaults(
...
+            "Track defaults",
+            "Processor latency or trim",
+            "Manual processor latency value used for new tracks, in frames.",
+        )
+        .category_order(10)
+        .setting_order(60)
+        .effect(SettingEffect::NextUse)
+        .editor(SettingEditor::SignedInteger {
+            min: -crate::MAX_TRACK_LATENCY_FRAMES,
+            max: crate::MAX_TRACK_LATENCY_FRAMES,
```
[the remark this answers is not in the document]
Comment (SanderVocke) [reply src/rust/shoop_egui/src/app_widget.rs:250]: Addressed in f1a4849b. The settings registry now supports cross-field draft validators, and the track-default registration rejects a negative processor frame value when the adjustment mode is `manual`. Negative values remain valid for `automatic_plus_trim`. A focused test covers both combinations.

File: src/rust/shoop_egui/src/app_widget.rs [hunk 4]
```diff
@@ -1719,16 +2004,98 @@ impl AppWidget {
...
+
+    fn resolve_track_default_saves(
+        &mut self,
+        state: &AppState,
+        settings_actions: &mut Vec<SettingsAction>,
+    ) {
+        for result in state.track_creation_results.iter() {
+            if let Some(draft) = self.pending_track_defaults.remove(&result.request_id) {
+                if result.success {
+                    settings_actions.push(SettingsAction::Save(draft));
```
[the remark this answers is not in the document]
Comment (SanderVocke) [reply src/rust/shoop_egui/src/app_widget.rs]: Addressed in f1a4849b. Confirmed defaults now remain queued while settings persistence is busy. Their save uses a correlated `SaveTrackDefaults` action; `Saving` and `StaleRevision` rejections retain the pending values, and the next attempt rebases only the six track-default fields onto the latest active settings revision. The pending entry is cleared after the settings manager accepts the save (or reports a terminal error). The updated test covers busy deferral, stale/rejected retention, rebasing, retry, and acceptance.
